### DISCUSSION TITLE
Embedding informant within existing Quarto website rather than linking out and losing sidebar etc.

### DISCUSSION THREAD

**Discussion by Sheron156** [rstudio/pointblank#541] [General]
Hi there,

I would like to have my informant object render within an existing quarto codebook/site rather than a link that takes them out of the book environment (would like sidebar w/search bar etc intact). I've tried generating the informant in a codeblock and then executing the object, and it looks great--except there are two versions of each variable/section. As if it's rendering twice...? I've disabled Quarto's table formatting but am not sure what my next steps could be. Maybe there's a handy function that I'm unaware of?--apologies if there's a simple solution that I somehow missed!! 

![image](https://github.com/rstudio/pointblank/assets/134512667/48a97ee3-a67a-4d27-bea7-bba74023d8f1)

**Comment by yjunechoe** [comment]
Are you on the latest version of pointblank? I *think* this should have been fixed by a recent PR #520. If you still have problems after updating, a reprex would be nice!

**Reply by Sheron156** [reply]
Dohhhh 🤦 ! I thought I'd grabbed the dev version so assumed I had the most recent version--oops. Thanks for the heads up. All updated and it's alllmost working perfectly now.  Just one little formatting issue. I could maybe modify the yaml output of the informant object to explicitly call the break?

Here is how the code below renders in the code block itself:

![image](https://github.com/rstudio/pointblank/assets/134512667/8304a418-8ba4-44fa-8fd2-58b9175ceaa2)

and when the qmd page is rendered--all good except section formatting is lost; all is in one blob rather than each category (info, range, etc.) on a new line. 

![image](https://github.com/rstudio/pointblank/assets/134512667/235626cf-24c2-499c-ae97-58bbd897019a)


``` r
library(pointblank)

# Create a pointblank `informant` object, with the
# tibble as the target table. Use a few information
# functions and end with `incorporate()`. The informant
# will then show you information about the tibble.
informant <- 
  dplyr::tibble(
    a = c(5, 7, 6, 5, NA, 7),
    b = c(6, 1, 0, 6,  0, 7)
  ) %>%
  create_informant(
    label = "A very *simple* example.",
    tbl_name = "example_tbl"
  ) %>%
  info_tabular(
    description = "This two-column table is nothing all that
    interesting, but, it's fine for examples on **GitHub**
    `README` pages. Column names are `a` and `b`. ((Cool stuff))"
  ) %>%
  info_columns(
    columns = a,
    info = "This column has an `NA` value. [[Watch out!]]<<color: red;>>"
  ) %>%
  info_columns(
    columns = a,
    info = "Mean value is `{a_mean}`."
  ) %>%
  info_columns(
    columns = b,
    info = "Like column `a`. The lowest value is `{b_lowest}`."
  ) %>%
  info_columns(
    columns = b,
    info = "The highest value is `{b_highest}`."
  ) %>%
  info_snippet(
    snippet_name = "a_mean",
    fn = ~ . %>% .$a %>% mean(na.rm = TRUE) %>% round(2)
  ) %>%
  info_snippet(snippet_name = "b_lowest", fn = snip_lowest("b")) %>%
  info_snippet(snippet_name = "b_highest", fn = snip_highest("b")) %>%
  info_section(
    section_name = "further information", 
    `examples and documentation` = "Examples for how to use the
    `info_*()` functions (and many more) are available at the
    [**pointblank** site](https://rstudio.github.io/pointblank/)."
  ) 

informant
```
<sup>Created on 2024-06-21 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

Thanks so much for your help! I absolutely love this package.

**Reply by yjunechoe** [reply]
Thanks for these details! It looks like there are a couple places where custom Quarto css styles are "leaking" into pointblank reports, some more visually breaking than others. I think we'll just have to address this in a separate, larger issue (pointblank is _just_ starting to catch up to quarto).

**Reply by Sheron156** [reply]
>  (pointblank is just starting to catch up to quarto)

same here tbh! :)

As to the custom styles... hrmmm! I'll take a look at what I have in my scss that could be messing with things.

I played around a bit before moving on to another project and found that if I wrote the informant as a yaml object, I could add html breaks to get it to do what I want--though of course not ideal with >100 variable codebook like mine. Another dirty fix that seemed to work--though I didn't test it thoroughly was modifying the check_info_columns_tbl function w/a break in the col name which would be a quicker work around than editing the yaml:

``` r
description_col_from_tbl <- function(x, tbl, .add = TRUE) {
    tbl <- check_info_columns_tbl(tbl = tbl)
    for (i in seq_along(tbl$column)) {
        x <- info_columns(x = x, columns = tbl$column[i], `Info <br>` = tbl$info[i], 
            .add = .add)
    }
    x
}
```

<sup>Created on 2024-06-24 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>


Thanks for looking into this! Will be going back to the project in a week or two and will dig around in the custom scss/css to see what I can find then.
